src/components/shared/CallNodeContextMenu.js

```diff
@@ -63,13 +63,18 @@ type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
 require('./CallNodeContextMenu.css');
 
 class CallNodeContextMenu extends PureComponent<Props> {
+  _hidingTimeout: TimeoutID | null = null;
 
-  _showMenu = () => {
+  _onShow = () => {
+    clearTimeout(this._hidingTimeout);
     this.props.setContextMenuVisibility(true);
   };
 
-  _hideMenu = () => {
-    this.props.setContextMenuVisibility(false);
+  _onHide = () => {
+    this._hidingTimeout = setTimeout(() => {
+      this._hidingTimeout = null;
+      this.props.setContextMenuVisibility(false);
+    }, 100);
```

Comment:
The value "100" is pretty much arbitrary but I found it to work fine. Maybe it could be lowered down to 50 or something.

Reply:
Blocking: I'm very scared of this `setTimeout`. I'm not going to "request changes" for this, especially since I'm out next week. But, I'd prefer either these two courses of actions:

1. Change this to properly hook into the right life cycle of the context menu visibility, if it is a setTimeout, then reference the commit in a comment, and point to the react-contextmenu code.
2. Add more code comments justifiying this, and probably a test.

Reply:
The tests all look like they run the timers, so there is at least that.

Reply:
Yeah I took really care at testing this in tests.
I really thought I added good comments to be honest. Probably I just kept them in my stupid brain... I totally agree this needs more explanation.

Reply:
```
// Using setTimeout here is a bit complex, but is necessary to make the menu
// work fine when we want to display it somewhere when it's already open
// somewhere else.
// This is the order of events in such a situation:
// 0. The menu is open somewhere, it means the user right clicked somewhere
 //     previously, and as a result some node has the "right clicked" status.
// 1. The user right clicks on another node. This is actually happening in
//    several events, the first event is "mousedown": this is where our own
//    components react for right click (both our TreeView and our charts)
//    and thus this is when the "right clicked" item is set in our store. BTW
//    this triggers a rerender of this component.
// 2. Then the event "mouseup" happens but we don't do anything for it for right
//    clicks.
// 3. Then the event "contextmenu" is triggered. This is the event that the
//    context menu library reacts to: first it closes the previous menu, then
//    opens the new one. This means that `_onHide` is called first for the
//    first menu, then `_onShow` for the second menu.
//    The problem here is that the call to `setContextMenuVisibility` we do in
//    `onHide` resets the value for the "right clicked" item. This is normally
//    what we want when the user closes the menu, but this case where the menu
//    is still open but for another node, we don't want to reset this value
//    which was set earlier when handling the "mousedown" event.
//    To avoid this problem we use this `setTimeout` call to delay the reset
//    just a bit, just in case we get a `_onShow` call right after that.
```

And actually I removed the "100" arbitrary value, as this works fine with just 0 because everything else is synchronous (there could be a non-synchronous case theorically but we don't use it).